### PR TITLE
Win32 error cleanup

### PR DESCRIPTION
**Pull request by AlexIljin**
Hello, dear readers of my full-pull blog!

  In this issue I'd like to unify the throwing of win32 errors in different parts of code, and also use the existing INVALID_HANDLE_VALUE constant instead of the "-1 <alien>" that some people came up with.

  For some reason, applying the last commit in this series leads to gdi+-error being thrown an startup. I, for the life of me, can't track the reason. Can you help me?

### CODE REVIEW

File: basis/windows/errors/errors.factor [hunk 1]
```diff
@@ -758,11 +755,7 @@ CONSTANT: expected-io-errors
     expected-io-errors member? ;
 
 : expected-io-error ( error-code -- )
```
Comment (mrjbq7) [inline basis/windows/errors/errors.factor]: this is weird that it drops the error code and throws ``GetLastError``.... is that the correct?

**Comment by AlexIljin** [comment]
I have added some more patches, but the code is not ready for review yet. I think I'll rearrange the commits and give better log messages.
I pushed it now just to have the work on GitHub, and so that you could see where I'm going with this.

**Comment by mrjbq7** [comment]
I wonder if you should simplify the words into two layers. One that gets last error and then defers to another word that checks and throws based on the error code.  We seem to have some duplication between ``n>windows-error-check`` and ``windows-error<>0`` for example.

**Comment by mrjbq7** [comment]
Also, not a big fan of ``n>`` as a word prefix, it's a little meaningless :-)

**Comment by erg** [comment]
I think it's pretty descriptive when you constrain ``n`` to always be an integer.

**Comment by mrjbq7** [comment]
What's the status of this PR?  Ready to merge?

**Comment by AlexIljin** [comment]
No, it's on pause. I'll notify when ready.

**Comment by catb0t** [comment]
did you end up not finding these changes useful? maybe if they're merged they can clean up other windows code we now have (well, aside from merging being a lot of work)

**Comment by AlexIljin** [comment]
I kind of lost interest in this work, though hoping to finish it some day.
Some of these changes I think are valuable, like replacing `-1 <alien>` with the `INVALID_HANDLE_VALUE` constant, because they make code more canonical and readable to a WinApi developer.
I think I need someone to help me with this.

**Comment by AlexIljin** [comment]
Mostly I wanted to reduce duplication and make all words for a neat tree.

**Comment by bjourne** [comment]
Looks like fine changes to me. Although some unit tests probably would be useful.

**Comment by mrjbq7** [comment]
@AlexIljin this seems like it's ready to merge as is?

**Comment by AlexIljin** [comment]
@mrjbq7, most of it is OK to merge, but the last patch throws up the `gdi+-error` when you try to `refresh-all`, and I still couldn't figure out why.

**Comment by mrjbq7** [comment]
Maybe it's an issue with ``refresh-all`` and you can still bootstrap into a good VM?

**Comment by mrjbq7** [comment]
Merged!

Let's see what it looks like on the build bots.

**Comment by mrjbq7** [comment]
(For certain changes that affect the UI on macOS and Linux, I sometimes have to do a console listener ``refresh-all save`` then start the UI.  Maybe that's required here.)

**Comment by AlexIljin** [comment]
I did `save` and restarted, then I saw this message on startup.

**Comment by mrjbq7** [comment]
Ok, cool. Will find and fix!

**Pull request closed without merging**